src/Materials/Node/Blocks/gradientBlock.ts

```diff
@@ -6,25 +6,57 @@ import { NodeMaterialBlockTargets } from '../Enums/nodeMaterialBlockTargets';
 import { _TypeStore } from '../../../Misc/typeStore';
 import { Color3 } from '../../../Maths/math.color';
 import { Scene } from '../../../scene';
+import { Observable } from '../../../Misc/observable';
 
 /**
  * Class used to store a color step for the GradientBlock
  */
 export class GradientBlockColorStep {
+    private _parent: GradientBlock;
+
+    private _step: number;
+    /**
+     * Gets value indicating which step this color is associated with (between 0 and 1)
+     */
+    public get step(): number {
+        return this._step;
+    }
+
+    /**
+     * Sets a value indicating which step this color is associated with (between 0 and 1)
+    */
+    public set step(val: number) {
+        this._step = val;
+        this._parent.onValueChangedObservable?.notifyObservers(this._parent);
+    }
+
+    private _color: Color3;
+
+    /**
+     * Gets the color associated with this step
+     */
+    public get color(): Color3 {
+        return this._color;
+    }
+
+    /**
+     * Sets the color associated with this step
+     */
+    public set color(val: Color3) {

```

Comment:
React loves immutable objects. It keeps the state auto-updatable and make your state management a bit easier. If you use these objects in a react state (as it seems to me, but please correct me if I am wrong!!) don't add a setter. instead create a new object.

Of course - unless it is impossible due to the architecture (i saw that the parent is being notified)

Reply:
will get resolved in a second PR.